qiita_db/support_files/patches/68.sql

```diff
@@ -0,0 +1,102 @@
+    -- inserting value per value of the table, this might take forever
+    FOR sid IN
+      EXECUTE 'SELECT sample_id FROM qiita.' || dyn_table_bk
+    LOOP
+      EXECUTE 'INSERT INTO ' || dyn_table || ' (sample_id, sample_values) VALUES (''' || sid || ''',  row_to_json((SELECT r FROM (SELECT * from qiita.' || dyn_table_bk || ' where sample_id = ''' || sid || ''') r))::jsonb - ''sample_id'');';
+    END LOOP;
+
+    -- adding index
+    EXECUTE 'ALTER TABLE ' || dyn_table || ' ADD CONSTRAINT pk_jsonb_' || dyn_t || ' PRIMARY KEY ( sample_id );';
+  END LOOP;
+END $do$;
+
+
+-- Now, let's move the data for the prep templates
+DO $do$
+DECLARE
+    dyn_t varchar;
+    dyn_table varchar;
+    dyn_table_bk varchar;
+    sid varchar;
+BEGIN
+  FOR dyn_t IN
+    SELECT DISTINCT table_name
+    FROM information_schema.columns
+    WHERE SUBSTR(table_name, 1, 5) = 'prep_'
+        AND table_schema = 'qiita'
+        AND table_name NOT IN ('prep_template',
+                               'prep_template_preprocessed_data',
+                               'prep_template_filepath',
+                               'prep_columns',
+                               'prep_template_processing_job',
+                               'prep_template_sample')
+  LOOP
+    dyn_table := 'qiita.' || dyn_t;
+    dyn_table_bk := dyn_t || '_bk';
+
+    -- rename the tables so we can move the data later
+    EXECUTE format('ALTER TABLE %1$s RENAME TO %2$s', dyn_table, dyn_table_bk);
+
+    -- create the new table, note that there are no constraints so the
+    -- inserts go fast but we will add them later
+    EXECUTE format('CREATE TABLE %1$s (sample_id VARCHAR NOT NULL, sample_values JSONB NOT NULL)', dyn_table);
+
+    -- inserting our helper column qiita_sample_column_names, which is gonna keep all our columns; this is much easier than trying to keep all rows with the same values
+    EXECUTE 'INSERT INTO ' || dyn_table || ' (sample_id, sample_values) VALUES (''qiita_sample_column_names'',  (''{"columns":'' || (SELECT json_agg(column_name::text) FROM information_schema.columns WHERE table_name=''' || dyn_table_bk || ''' AND table_schema=''qiita'' AND column_name != ''sample_id'')::text || ''}'')::json);';
+
+    -- inserting value per value of the table, this might take forever
+    FOR sid IN
+      EXECUTE 'SELECT sample_id FROM qiita.' || dyn_table_bk
+    LOOP
+      EXECUTE 'INSERT INTO ' || dyn_table || ' (sample_id, sample_values) VALUES (''' || sid || ''',  row_to_json((SELECT r FROM (SELECT * from qiita.' || dyn_table_bk || ' where sample_id = ''' || sid || ''') r)));';
+    END LOOP;
+
+    -- adding index
+    EXECUTE 'ALTER TABLE ' || dyn_table || ' ADD CONSTRAINT pk_jsonb_' || dyn_t || ' PRIMARY KEY ( sample_id );';
+  END LOOP;
+END $do$;
+
+-- Dropping all the _bk tables
```

Comment:
Confirmed DROP TABLES will free space on COMMIT, provided there are no other connections using the table as well. The OS should reclaim the space in time, although it may not be immediately after performing the operation; hence, an exceedingly large number of _bk tables being created and dropped may impact system resources, fyi.